### PR TITLE
[Yaml] fix inline handling when dumping tagged values

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #32251
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Yaml/Tests/DumperTest.php [hunk 1]
```diff
@@ -434,6 +435,98 @@ public function testDumpingStdClassInstancesRespectsInlineLevel()
...
+            ]),
+            'user2' => new TaggedValue('user', [
+                'username' => 'john',
+            ]),
+        ];
+
+        $yaml = $this->dumper->dump($data, 1);
+
+        $expected = <<<YAML
+user1: !user
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Yaml/Tests/DumperTest.php]: do we want to put inline object on the same line as their tag?

Comment (xabbuh) [reply src/Symfony/Component/Yaml/Tests/DumperTest.php]: sounds like a good idea

Comment (xabbuh) [reply src/Symfony/Component/Yaml/Tests/DumperTest.php]: done

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**